src/script/lua_api/l_particles.cpp

```diff
@@ -38,6 +39,35 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 // texture = e.g."default_wood.png"
 // animation = TileAnimation definition
 // glow = num
+
+void LuaParticleParams::readTexValue(lua_State* L, ServerParticleTexture& tex) {
+	lua_Integer top = lua_gettop(L);
+	tex.animated = false;
+	if (lua_isstring(L, -1)) {
+		tex.string = lua_tostring(L, -1);
+	} else if (lua_istable(L, -1)) {
+		lua_getfield(L, -1, "img");
+		if (lua_isstring(L, -1)) {
+			tex.string = lua_tostring(L, -1);
+		} else goto error;
+		lua_pop(L, 1);
+
+		lua_getfield(L, -1, "animation");
+		if (! lua_isnil(L, -1)) {
+			tex.animated = true;
+			tex.animation = read_animation_definition(L, -1);
+		}
+		lua_pop(L, 1);
+
+		LuaParticleParams::readTweenTable(L, "alpha", tex.alpha);
+		LuaParticleParams::readTweenTable(L, "scale", tex.scale);
+	} else goto error;
+	lua_settop(L, top);
+	return;
+
+	error: lua_pushliteral(L, "invalid type in texture specifier"), lua_error(L);
```

Comment:
To throw an exception you can do this
```suggestion
	throw LuaError("invalid type in texture specifier");
```

But instead of the explicit type checks I'd use `luaL_checktype`